lib/handlers/subscribe.js

```diff
@@ -60,6 +60,21 @@ function storeSubscriptions (sub, done) {
   var broker = client.broker
   var perst = broker.persistence
 
+  if (sub && packet.subscriptions[0].topic !== sub.topic) {
+    // don't call addSubscriptions per topic,
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    return done(null, sub)
+  }
+
+  if (!packet.messageId) {
+    // this is a restoreSubscription, don't addSubscriptions!
+    // TODO change aedes subscribe handle, but this is a fast & dirty fix for now
+    if (perst._addedSubscriptions) {
+      perst._addedSubscriptions(client, packet.subscriptions)
+    }
```

Comment:
Problem is `persistence#addSubscriptions` should be called only when new subscribes arrives, but it is currently always called (also on restore subscriptions)
Another issue is that, any logic related to subscription handling (like any children override logics as inside `_addedSubscriptions`) should not go inside `addSubscriptions`, since in both cases (new sub and restore sub) children may decide to run that logics.

So, I've also another proposal @mcollina 
aedes subscribe handler emits a `clientSubscription` event, so that persistence children can trigger and call their logic there. in this case `_addedSubscriptions` would be called on that listener inside aedes-cached-persistence.